### ISSUE TITLE
List and document experimental features in one place

### ISSUE THREAD

**Issue by fricklerhandwerk** [NixOS/nix#8466]
## Problem

Currently experimental features are listed in the setting `experimental-features`, but documented in detail on a rather explanation-style page aimed at contributors. 

## Proposal

The second should not have that listing at all because it's unrelated, and the listing in the setting should contain the details.

## Checklist

- [x] checked [latest Nix manual] \([source])
- [x] checked [open documentation issues and pull requests] for possible duplicates

[latest Nix manual]: https://nixos.org/manual/nix/unstable/
[source]: https://github.com/NixOS/nix/tree/master/doc/manual/src
[open documentation issues and pull requests]: https://github.com/NixOS/nix/labels/documentation

## Priorities

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by abathur** [comment]
I guess this might be a good place to address how mature/safe the experimental features are to play with? Some rough proxy for whether novices should avoid it and whether it's ripe for ~playtesting?

**Comment by fricklerhandwerk** [comment]
Sure, would be good to have that, but this is a different issue.

**Comment by b-m-f** [comment]
@fricklerhandwerk can you point me to the code where I can find the
1. setting where details should be added
2. contributor page where details should be removed from

**Comment by fricklerhandwerk** [comment]
Hey @b-m-f, thanks for giving it a stab!

You'll find the explanation page with `fd experimental **/*.md`. The details for each experimental feature are already there, they're just rendered into the same page. Follow the replacement strings from that file with `rg`. Check the build setup in `doc/manual/local.mk` for how it's wired up (it's a bit tricky...).

**Comment by Ericson2314** [comment]
I disagree that the settings page should describe each experimental feature. The settings page is already super big, and I don't think it should contain much more than brief descriptions and links to larger discussions elsewhere.

Basically "global settings" is a misfeature; it is a bad jumbled of unrelated concerns in both the docs and the code. Having it act more like an index to topics discussed in greater depth elsewhere can be seen as a prelude to https://github.com/NixOS/nix/issues/5638.

All that said, I do agree that we shouldn't describe the experimental features themselves in the same contribution guide page we talk about the experimental process itself. I do support separating that out as the issue describes.

**Comment by fricklerhandwerk** [comment]
Breaking up the settings page is a topic on its own. But for now the most reasonable thing would be to let the listing of experimental features live next to the setting because this is what it's most closely related to.

**Comment by Ericson2314** [comment]
I agree breaking up settings can wait, but I disagree that means we should stuff the settings pages full of more stuff in the meantime. We can simply create other pages with actually long-form text and have the settings page link to them.

**Comment by fricklerhandwerk** [comment]
Okay, which cheap, low-tech options do we have here? The settings page has the advantage that it's accessible as a `man` page and I wouldn't want to give up on that for experimental features.

What if we had a page per experimental feature that we could link to and from all the bits of interface that belong to it? Then we can have some space for extensively documenting trade-offs and such. And we could render a `man` page for each, like man nix-experimental-<feature flag>`.

In the web rendering we could nest the feature pages under the current explanation page.
